ecs-init/docker/docker.go

```diff
@@ -541,3 +576,32 @@ func (c *client) StopAgent() error {
 	}
 	return err
 }
+
+// isDomainJoined is used to validate if container instance is part of a valid active directory.
+func isDomainJoined() bool {
+	realmPath, err := exec.LookPath("realm")
+	if err != nil {
+		log.Errorf("realmd is not installed on the instance")
+		return false
+	}
+
+	if len(realmPath) == 0 {
+		log.Errorf("could not path of realm on the instance")
+		return false
+	}
+
+	realmCommmand := realmPath + " list"
+	realmCmdOutput, err := exec.Command("bash", "-c", realmCommmand).Output()
+	if err != nil {
+		log.Errorf("failed to read realm info")
+		return false
+	}
+
+	realmOutputStr := strings.TrimSpace(string(realmCmdOutput))
```

Comment:
nit: do you need to do `TrimSpace` if you're doing a `string.Contains` on `realmOutputStr`?